xml_analyzer/xmlreader.py

```diff
@@ -1,38 +1,37 @@
 import collections
 import xml.etree.ElementTree as Et
 from .data_container import DataContainer
+from .definitions import AutocalcMethods
 
 
 class XMLReader:
-    autocalc_methods = ['autosum']
 
     def __init__(self, project_file):
         self._project_file = project_file
         self.tree = Et.parse(self._project_file)
         self.root = self.tree.getroot()
-        self.pv_map = collections.namedtuple('property_value_mapping', ['property', 'value'])
 
     @staticmethod
-    def parse_constraint_spec(constraint_spec: str) -> list:
+    def _parse_constraint_spec(constraint_spec: str) -> list:
         autocalc_sysml_type_name = list()
-        for method in XMLReader.autocalc_methods:
-            if method in constraint_spec:
-                autocalc_sysml_type_name.append(method)
-                for occurrence in constraint_spec.split(method)[1:]:
+        for method in AutocalcMethods:
+            if method.name in constraint_spec:
+                autocalc_sysml_type_name.append(method.name)
+                for occurrence in constraint_spec.split(method.name)[1:]:
                     search_param = occurrence[occurrence.find('(') + 1:occurrence.find(')')]
                     if search_param != '':
                         autocalc_sysml_type_name.append(search_param)
         return autocalc_sysml_type_name
 
-    def find_attributes(self, val: list, method='by_type'):
+    def _find_attributes(self, val: list, method='by_type'):
         elements = None
         if method == 'by_type':
             query = ".//Attribute/Type//*[@Name='{}']/../..".format(val)
             elements = self.root.findall(query)
 
         return elements
 
-    def resolve_bindings(self, constraint_property: Et.Element) -> list:
+    def _resolve_bindings(self, constraint_property: Et.Element) -> list:
```

Comment:
Instead of using list as type hint, you can use the metatype that was built for type hints.

`from typing import List`

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/125a9de3f5317b3cc4f907790a9fabccc9134c0e